### ISSUE TITLE
Editor "zoom" (view scaling without changing document font size)

### ISSUE THREAD

**Issue by 20goto10** [vkbo/novelWriter#1280]
I've been searching for a way to scale the view window for the editor without affecting the output of the manuscript. If the option is there in the settings, I can't find it. I have gotten by by adjusting the editor font size instead, but whenever I render the manuscript I wind up having to shift everything down from 19 point to standard 12 point text. This problem may only be apparent on ultra high def laptop screens.

How difficult is it  to add a "zoom" option like this, along the lines of what exists in Word/LibreOffice/etc? I'd rather avoid having to mess with external settings, i.e. display, scaling my window manager, etc..

**Comment by vkbo** [comment]
Hmm, it should auto-scale for high DPI. Can you provide a screenshot? This may be a bug.

I have considered adding a zoom feature, but I haven't investigated it yet.

**Comment by 20goto10** [comment]
> Hmm, it should auto-scale for high DPI. Can you provide a screenshot? This may be a bug.
> 
> I have considered adding a zoom feature, but I haven't investigated it yet.

Here's how it looks for me in the basic mode and focus mode. Font size is 12 point here. The screen is 14", so, pretty tricky to use on the default. I think automatically scaling the view to the screen size is not ideal as it does not take personal reading preferences into account (not to mention more specific accessibility issues).

![Screenshot from 2022-12-05 13-31-07](https://user-images.githubusercontent.com/3967045/205714872-0c91831b-e1e1-44d1-940d-09a3cb0067a4.png)
![Screenshot from 2022-12-05 13-31-16](https://user-images.githubusercontent.com/3967045/205714891-c578d50c-c656-434a-8ec6-febca42ad705.png)

I _have_ seen an issue where the font sizes and line spaces all shrink after my screen saver activates, but for some reason I assumed that was something to do with my window manager and not NovelWriter itself. (I'm using Mint Linux 21 w/Cinnamon, which has a few issues of its own.) That's fixed by restarting NovelWriter. Maybe it is a function of the automatic scaling... if so, it's going in the wrong direction. I see it as a separate issue from the matter of zoom.

**Comment by 20goto10** [comment]
(I should also note, system font size is 12 and there is no WM font scaling in effect.)

**Comment by vkbo** [comment]
> Here's how it looks for me in the basic mode and focus mode. Font size is 12 point here. The screen is 14", so, pretty tricky to use on the default. I think automatically scaling the view to the screen size is not ideal as it does not take personal reading preferences into account (not to mention more specific accessibility issues).

The auto-scaling is not to screen size, but to the GUI scaling factor. If you have it set to 100% (1.0), then there is no scaling. The font is scaled automatically by the Qt framework anyway. All I do is scale boxes and icons to match.

I don't think that's the issue here.

> I _have_ seen an issue where the font sizes and line spaces all shrink after my screen saver activates, but for some reason I assumed that was something to do with my window manager and not NovelWriter itself. (I'm using Mint Linux 21 w/Cinnamon, which has a few issues of its own.) That's fixed by restarting NovelWriter. Maybe it is a function of the automatic scaling... if so, it's going in the wrong direction. I see it as a separate issue from the matter of zoom.

The auto-scaling is not dynamic, so if there is something glitchy it is likely an issue between the Qt framework and the window manager. Qt5 has a few issues in this respect, and there isn't really anything I can do about it. Either they fix it in patch releases, or hopefully I can add support for Qt6 eventually. That should help with a bunch of little issues with modern window managers and High-DPI monitors.

In any case, I still don't quite understand what the issue is with font size. The GUI font size, editor font size, and the font size output by the Build tool to manuscript documents are all independent settings, and you should be able to tweak each as you want.

Unrelated, but I also see that you are running a light theme icons on a dark GUI theme, which makes the icons hard to see. Especially the black on grey ones. You should switch to the dark icon theme. This also seems to be an older version of novelWriter.

**Comment by vkbo** [comment]
As for the zoom feature, the Qt framework does have a text zoom feature for the editor, but unfortunately the implementation isn't very useful. Zooming just changes the base text size, so the headers don't scale. It can theoretically be fixed by setting relative sizes for the headers, but that too is rather limited as you can't set numeric relative sizes. It's one of the problems I've tried to solve since I first started this project.

**Comment by 20goto10** [comment]
Fair enough. I see what you mean about Qt problems and I understand the reasons for not having a zoom feature now. I completely overlooked the font size setting in the export screen--except once, I guess, because at some point I must have set it to 19 point (oops). Cinnamon/Gnome stuff has a separate Qt settings editor which made some of the smaller Qt-controlled content a little easier to read, too. So I have everything in good shape now (and I've now updated to 2.0). It seems I can live without zoom. 

Thanks for the guidance and your work on this excellent project. I've been recommending it to many writers.

**Comment by vkbo** [comment]
I added those separate settings as a way to compensate for the lack of zoom, at least in the context of usability. There are still cases where the lack of zoom is an issue, so I would like to fix it. I just don't have a solution right now that doesn't come with other issues. I would like to look for a workable solution though, so let's leave this issue open for now.

Thanks a lot for your feedback! I'm happy that novelWriter works for you aside from this. It's nice to see that it is useful to other people. I originally made it for myself and a colleague, and it has grown to quite a large project.

**Comment by julichan** [comment]
Hello,

I'm coming for update on this. I'm on windows and i see the view rescales only upon application restart after changing the scaling values in windows settings. I'm also totally for a manual zoom and automatic update upon change. I'm not experienced with QT technologies so i don't know about its limitations but isn't there just a way to regenerate the whole UI when that setting has been changed in windows or overriden in the setting of the application ? Is there any problems to implement a simple solution to begin with ? or is it just wanted to have something completely dynamic to begin with ?

Thanks.

**Comment by vkbo** [comment]
I have this on my short list. Zoom for the text in the documents is doable. The only current issue is the headings don't scale, but that is solvable. Scaling the GUI itself is a very different matter and will still require an app restart.

This is a widget-based app, not web tech, so changing the geometry of everything is non-trivial after it's drawn. Qt can change some things dynamically when settings change, but a lot of things do not change. It's a lot more than just changing the font. A lot of GUI elements are scaled proportionally to the font. To allow zooming the app itself, it would require dynamically scaling thousands of little values in a ton of places. That is a very tall order for such a minimal benefit of not having to restart the app.

Scaling of the document text is coming, but not the GUI itself.

**Comment by sjuswede** [comment]
Not the exact same issue, but it appears to be related;

I run NW on a small laptop (10") with a FHD display. When I first started it, everything was super tiny, but I increased font sizes and that fixed just about all problems. There were some rather cramped initial dialog sizes, but after resizing them they for the most part seem to remember that they need to be larger to be usable.

Except for the font selection dialouges. They start up postage stamp sized, and using the tiny touchpad to resize them is a real pain. It doesn't seem to matter whether I'm using the Qt or native dialog setting, they're still super tiny, and they don't respect my WM's commands to full size (I use niri). That's likely a Qt thing, but it does make for a painful experience using them.

**Comment by vkbo** [comment]
Hi @sjuswede, this issue annoys me too to be honest. The font dialog is handled entirely by Qt, as are all the file dialogs. It is unrelated to the zoom, and I've tested before if it is possible to resize them to their previously known size, and it is with the Qt style dialog, but it ignores the setting if it shows the native dialog. I suppose that would be an improvement nonetheless, so feel free to create a new issue ticket about this and I'll add it to the next release. It isn't particularly difficult to add.

**Comment by DiiaBlood** [comment]
Hello! Adding to this too! I would love more zooming control in in more areas in the editor. My eyes are quite bad!

**Comment by vkbo** [comment]
You can change the font size of the entire app in Preferences if it is too small for general use. This is a better option than zooming in those cases. Zoom will only affect the editor/viewer.

**Comment by soratoyuki** [comment]
Also seconding a Zoom ability independent of font size. Full-screening the app leads to a lot of empty space in the main writing section. This could be a user issue, though.

<img width="2560" height="1440" alt="Image" src="https://github.com/user-attachments/assets/7213eb20-7516-43f6-a17f-441e6ad83696" />

**Comment by vkbo** [comment]
You can increase both the width of the text area and the font size in Preferences.

**Comment by FrostKnight** [comment]
i would like this as well, it would very much help me when I have someone look over my work. So I don't need to load it on my tv.

**Comment by FrostKnight** [comment]
> You can change the font size of the entire app in Preferences if it is too small for general use. This is a better option than zooming in those cases. Zoom will only affect the editor/viewer.

Where do you do this in?

**Comment by vkbo** [comment]
> Where do you do this in?

Change "Application font" in Preferences for the app, and "Document font" for the editor/viewer.
